### PR TITLE
Speed up `_check_{unit}_range()` functions

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

The inputs of `_check_hour_range()`, `_check_minute_range()` and `_check_second_range()` are always Python scalars, so it is faster to handle them in pure Python than with `numpy`. This speeds up `pytest astropy/coordinates` by 20% or so (i.e. by a few seconds on my computer). I also fixed some discrepancies between the behavior and the description of the functions, but those discrepancies do not seem to have caused any bugs visible to users.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Review by neutrinoceros — APPROVED** [review]
very nice, and I can also see the speedup just by running tests locally. Thank you !

**Review by mhvk — COMMENTED** [review]
These routines seem to be properly used in `Angle` itself, on the output from the string parser, where indeed floats are the only possible option, and in `formats.check_hms_ranges`, which is not used anywhere in astropy. In principle, that routine can presently be used with arrays. Should we just remove it?

File: astropy/coordinates/angles/formats.py [hunk 1]
```diff
@@ -319,52 +319,39 @@ def parse(self, angle, unit, debug=False):
...
 
-def _check_hour_range(hrs):
+def _check_hour_range(hrs: float) -> None:
     """
-    Checks that the given value is in the range (-24, 24).
+    Checks that the given value is in the range [-24,24].  If the value
+    is equal to -24 or 24, then a warning is raised.
     """
-    if np.any(np.abs(hrs) == 24.0):
+    if abs(hrs) == 24.0:
```
Comment (mhvk) [inline astropy/coordinates/angles/formats.py:327]: Probably really not worth optimizing this further, but in principle one would want OK hours to go through fastest, so I think the pattern one wants is,
```
if not -24.0 < hrs < 24.0:
    if abs(hrs) > 24.0:
        raise(...)
    warn(...)
```

**Review by mhvk — APPROVED** [review]
I think this is fine, though I still have a suggestion in-line (which you should feel free to ignore).

File: astropy/coordinates/angles/formats.py (hunk 1 above)
Comment (eerovaher) [reply astropy/coordinates/angles/formats.py]: Code for timing the function:
```
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.15.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from astropy.coordinates.angles.formats import _check_hour_range

In [2]: %timeit _check_hour_range(15)
```
Results on current `main`:
```
8.79 µs ± 22.6 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```
With my original patch:
```
79.8 ns ± 0.0881 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```
With your suggestion:
```
59.5 ns ± 0.118 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```
So the difference between my original patch and your suggestion is not large enough to be noticeable when running `pytest astropy/coordinates`, but your suggestion is faster.

**Comment by mhvk** [comment]
@pllim - another one stuck with readthedocs... Can we get your kind override?  Thanks!!

**Comment by pllim** [comment]
Done. Thanks!

**Pull request merged**